src/systemTest/java/com/flightstats/hub/model/ChannelConfigExpirationSettings.java

```diff
@@ -0,0 +1,73 @@
+package com.flightstats.hub.model;
+
+import lombok.Builder;
+import lombok.Value;
+import org.joda.time.DateTime;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Builder
+@Value
+public class ChannelConfigExpirationSettings {
+    String channelName;
+
+    boolean keepForever;
+    long ttlDays;
+    DateTime mutableTime;
+    long maxItems;
+
+    public Optional<DateTime> getMutableTime() {
+        return Optional.ofNullable(mutableTime);
+    }
+
+    public static class ChannelConfigExpirationSettingsBuilder {
```

Comment:
What do you think of lombok-ing this and moving the validation to another class that is the one that is actually consumed, one that essentially takes the arguments, creates an instance with a builder, then validates that. Hmm, then I guess that would have to extend this... NM

Reply:
Is there a custom validation @annotation out there? that would be nice. Found this https://dzone.com/articles/create-your-own-constraint-with-bean-validation-20

Reply:
This seemed to be the way that the lombok docs suggested validation be done. I'll see if I can find the particular page I was using as a guide... and I'll take a look at your link.

Reply:
I lied about the example being from Lombok maintainers. I might be able to override the actual `build()` function to do the validation there...? The constraint annotation route seems a little heavy-handed for this tiny use case, but I can likely be convinced otherwise.